src/resources/views/index.blade.php

```diff
@@ -49,38 +49,80 @@
       ]
     });
 
-    $('#notifications-driver-channels').on('show.bs.modal', function (event) {
+    $('#notifications-driver-channels')
+      .on('show.bs.modal', function (event) {
         $(this).find('span[rel="driver"]').text(event.relatedTarget.dataset.driver);
         $(this).find('span[rel="notification"]').text(event.relatedTarget.dataset.title);
         $(this).find('input[name="driver"]').val(event.relatedTarget.dataset.driver);
         $(this).find('input[name="notification"]').val(event.relatedTarget.dataset.notification);
 
+        // request a list of available channels to the driver
         $('#available-channels').select2({
+          ajax: {
```

Comment:
i'd like some loading spinner that indicates that the channels are being loaded (see the other comment)